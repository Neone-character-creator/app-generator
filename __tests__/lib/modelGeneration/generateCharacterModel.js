const modelValidation = require("../../../lib/modelValidation");

describe("model generation", () => {
    it("requires a model property", () => {
        const configuration = {};
        expect(() => {
            modelValidation(configuration);
        }).toThrowErrorMatchingSnapshot();
    });
    describe("character model", () => {
        it("must be defined, with at least one property", () => {
            expect.assertions(1);
            try {
                const configuration = {
                    model: {}
                };
                modelValidation(configuration)
            } catch (e) {
                expect(e).toMatchSnapshot();
            }
        });
        it("must have at least one property", () => {
            const configuration = {
                model: {
                    character: {
                    }
                }
            };
            expect(() => {
                modelValidation(configuration);
            }).toMatchSnapshot();
        });
    });
    describe("complex type", () => {
        describe("properties", () => {
            it("may be a string specifying their type", () => {
                const configuration = {
                    model: {
                        character: {
                            properties: {
                                name: "string",
                                age: "number"
                            }
                        }
                    }
                };
                modelValidation(configuration)
            });
            it("may reference another complex type", () => {
                const configuration = {
                    model: {
                        character: {
                            properties: {
                                career: "career"
                            }
                        },
                        career: {
                            properties: {}
                        }
                    }
                };
                modelValidation(configuration)
            });
            it("may be an array of 'number'", () => {
                const configuration = {
                    model: {
                        character: {
                            properties: {
                                age: "number"
                            }
                        }
                    }
                };
                modelValidation(configuration)
            });
            it("throws an error if a non-existent type is referenced", () => {
                const configuration = {
                    model: {
                        character: {
                            career: "fake"
                        },
                        real: {}
                    }
                };
                expect(() => {
                    modelValidation(configuration)
                }).toMatchSnapshot();
            });

        });
    });
});