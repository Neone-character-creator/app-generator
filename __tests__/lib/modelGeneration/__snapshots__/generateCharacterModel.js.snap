// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`model generation character model must be defined, with at least one property 1`] = `"received is not a function"`;

exports[`model generation character model must have a properties object 1`] = `"Element missing required properties object"`;

exports[`model generation complex type properties throws an error if a complex type has no properties object 1`] = `"Element missing required properties object"`;

exports[`model generation complex type properties throws an error if a non-existent type is referenced 1`] = `"Type was fake but must be one of character, real, number, string"`;

exports[`model generation complex type properties throws an error if a property is not a string 1`] = `"Properties can only be strings"`;

exports[`model generation requires a model property 1`] = `"model is a required field"`;
